### DISCUSSION TITLE
Caching DNS Resolver

### DISCUSSION THREAD

**Discussion by blakegonzales** [docker-mailserver/docker-mailserver#3722] [General]
Does DMS include a caching DNS resolver?
If not, would it be best practice to install `unbound` on the linux DMS host?

A lot of the services in DMS I think heavily depend on DNS requests .  Some service providers rate-limit access to their pre-defined DNS resolvers, which might cause trouble.

Thoughts anyone?

**Comment by polarathene** [comment]
I am aware of it being recommended for anti-spam services, others may chime in (_not something I have a strong opinion on_):
- https://github.com/docker-mailserver/docker-mailserver/issues/2918#issuecomment-1335287973
- https://github.com/orgs/docker-mailserver/discussions/2917
- https://docker-mailserver.github.io/docker-mailserver/v13.0/config/security/rspamd/#rbls-realtime-blacklists-dnsbls-dns-based-blacklists
- https://docker-mailserver.github.io/docker-mailserver/v13.0/config/environment/#enable_dnsbl

On a linux host, you could probably also use `systemd-resolved` for caching. If you're comfortable running `unbound` or similar instead that's fine too 👍

**Comment by blakegonzales** [comment]
Thank you for the thoroughly documented response!

I've been running `unbound` for a while now.. and I'll deep doing so ✅.

**Reply by casperklein** [reply]
Independently from running DMS, it's a good practice to run a local DNS cache IMO (to speed up things / reduce external DNS queries) 👍 I've been using bind9 for years, but any other software like unbound, PowerDNS, etc. is also suitable.
